kasa/smart/modules/time.py

```diff
@@ -83,3 +84,12 @@ async def set_time(self, dt: datetime) -> dict:
             if region:
                 params["region"] = region
         return await self.call("set_device_time", params)
+
+    async def _check_supported(self):
+        """Additional check to see if the module is supported by the device.
+
+        Hub attached sensors report the time module but do return device time.
+        """
+        if (parent := self._device.parent) and parent.device_type == DeviceType.Hub:
+            return False
```

Comment:
Looks like this check is done multiple times (should probably also be `device_type is DeviceType.Hub`?), perhaps it makes sense to add a `_is_hub_child()` helper?

Reply:
Done